plugins/gin/gin.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Tetrate Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package http contains several client/server http plugin which can be used for integration with net/http.
+*/
+
+package gin
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/tetratelabs/go2sky"
+	"github.com/tetratelabs/go2sky/propagation"
+	"github.com/tetratelabs/go2sky/reporter/grpc/common"
+	"strconv"
+	"time"
+)
+
+const (
+	httpServerComponentID int32 = 49
+)
+
+//Middleware gin middleware return HandlerFunc  with tracing.
+func  Middleware(tracer *go2sky.Tracer) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if tracer != nil {
+			operationName :=fmt.Sprintf("/%s%s", c.Request.Method, c.Request.URL.Path)
```

Comment:
URL.Path is the actual path of the request: e.g. `/my/path/123` this is going to be hight cardinality, is that intended? usually the route approach is preferred, e.g. `/my/path/{resource_id}` instead. Not sure how this can be achieved with gin but worstcase you can use the context.HandlerName (https://godoc.org/github.com/gin-gonic/gin#Context.HandlerName)

Reply:
Agree. Route approach is recommended, such as Java SpringMVC plugin.

Reply:
yes,you are right. will fix.

Reply:
Done.